src/openfluid/ui/waresdev/WareSrcExplorer.cpp

```diff
@@ -432,6 +439,41 @@ void WareSrcExplorer::onNewFragmentAsked()
 // =====================================================================
 
 
+void WareSrcExplorer::onFragmentRemovalAsked()
+{
+  QString Path = mp_Model->filePath(currentIndex());
+  openfluid::ui::waresdev::GitUIProxy Git;
+
+  openfluid::waresdev::WareSrcEnquirer::WarePathInfo WareInfo = 
+    openfluid::waresdev::WareSrcEnquirer::getWareInfoFromPath(Path.toStdString());
+
+  QString WarePath = 
+    QString::fromStdString(
+      WareInfo.AbsoluteWarePath
+    );
+
+  QString SubPath = 
+    QString::fromStdString(
+      WareInfo.RelativePathToWare
+    );
+
+  if (QMessageBox::warning(this, tr("Delete fragment"), tr("Are you sure you want to delete \"%1\"?").arg(SubPath),
+                           QMessageBox::Ok | QMessageBox::Cancel, QMessageBox::Cancel) == QMessageBox::Cancel)
+  {
+    return;
+  }
+  
+  openfluid::ui::waresdev::WareGitDialog Dialog;
+  Dialog.setWindowTitle("git rm submodule");
+  Dialog.setContent(Git.removeSubmodule(WarePath, SubPath));
+  Dialog.exec();
```

Comment:
This seems to broke SOLID principles, but this code is maybe inspired by other cases in the wares/frament management? to be reviewed